### ISSUE TITLE
Is the PUT method supported?

### ISSUE THREAD

**Issue by nathan-alden** [genotrance/px#6]
Hi, geno. I am using a continuous integration tool called Concourse that has a command line application known as Fly. Fly is a Linux tool, so it reads proxy settings from the `HTTP_PROXY`, `HTTPS_PROXY` and `NO_PROXY` environment variables. I am running Fly from within a Windows 10 VM running on my workstation.

Settings:

| Setting | Value |
| ------- | ----- |
| VM Host IP | `192.168.95.1` |
| VM Host px `port` | `3128` |
| VM Host px `gateway` | `1` |
| VM Host px `noproxy` | `127.0.0.0/24,192.168.95.0/24,10.<redacted>` |
| VM Guest HTTP_PROXY environment variable | `http://192.168.95.1:3128` |
| VM Guest HTTPS_PROXY environment variable | `http://192.168.95.1:3128` |
| VM Guest NO_PROXY environment variable | `127.0.0.0/24,localhost,.corporatenetwork.com` |
| Remote Concourse URL | `10.<redacted>` |

Note that `10.<redacted>` in both cases is the single IP address of the Concourse server on my corporate network.

When I run Fly in this manner, I receive the following strange HTTP response:
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
        "http://www.w3.org/TR/html4/strict.dtd">
<html>
    <head>
        <meta http-equiv="Content-Type" content="text/html;charset=utf-8">
        <title>Error response</title>
    </head>
    <body>
        <h1>Error response</h1>
        <p>Error code: 501</p>
        <p>Message: Unsupported method ('PUT').</p>
        <p>Error code explanation: HTTPStatus.NOT_IMPLEMENTED - Server does not support this operation.</p>
    </body>
</html>
```
However, if I simply add `10.<redacted>` to the `NO_PROXY` environment variable and try again, Fly works without issue. It seems like the only difference between my two attempts is that the first goes through px and the second doesn't.

Partial log output demonstrating that `GET` works fine but `PUT` doesn't:
```
192.168.95.129 - - [14/Mar/2017 08:40:33] "GET http://10.<redacted>:8080/api/v1/info HTTP/1.1" 200 -
...
192.168.95.129 - - [14/Mar/2017 08:40:35] code 501, message Unsupported method ('PUT')
192.168.95.129 - - [14/Mar/2017 08:40:35] "PUT http://10.<redacted>:8080/api/v1/<redacted> HTTP/1.1" 501 -
```
Does px support `PUT`? Do I need to configure something somewhere?

**Comment by nathan-alden** [comment]
I looked at your code and it seems you are missing support for some HTTP methods like `PUT` and `DELETE`.

**Comment by genotrance** [comment]
Yes, this was an oversight. I need to add these and verify that they work.

**Comment by nathan-alden** [comment]
Cool. I figured it was just that.

**Comment by genotrance** [comment]
This has been fixed in v0.2.1. I'll close it out once I hear from you that it works as expected.

**Comment by nathan-alden** [comment]
I'll test this in the next couple of days.

As an aside, you inspired me to write my own proxy in C#/.NET Core: https://github.com/nathan-alden/proxy. Thanks for the inspiration!
